From 4e21554c274425f450b24af1a0537b98eaaf922c Mon Sep 17 00:00:00 2001
From: dmeier <dmeier@90b65887-3827-0410-9a23-83215b262276>
Date: Wed, 25 May 2016 14:57:18 +0000
Subject: [PATCH 0752/1077] MANTIS 40178 new component/devicehandler for
 TTS(text to speech)

Added listener waiting for playComplete
removed argument blocking from play()

git-svn-id: svn://localhost/SelfServiceCommon/trunk@1021 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/AnnounceDev.idl         | 22 +++++++++++++------
 1 file changed, 15 insertions(+), 7 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/AnnounceDev.idl b/SelfServiceCommon/Massai/idl/Massai/AnnounceDev.idl
index e7c6bb09..1c33bea5 100644
--- a/SelfServiceCommon/Massai/idl/Massai/AnnounceDev.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/AnnounceDev.idl
@@ -4,7 +4,7 @@
 //
 //  purpose:  interface to announcement handler
 //
-//  date:     28.04.2016
+//  date:     25.05.2016
 //
 //  author:   dmeier
 //
@@ -30,15 +30,23 @@ module AnnounceDev
     RC_WRONG_STATE                      /**<  The handler is unable to execute the function*/
   };
 
+  interface AnnounceListener
+  {
+    void playComplete();                /**< play is complete, i.e. speaking finished or stopped */
+  };
+	
   interface AnnounceControl
   {
-    RC play( in boolean           blocking,
-             in bytestream        ssml );
-    RC stop(  );
+    RC play( in bytestream ssml );      /**< Play aka speak the text given in ssml */
+    RC stop(  );                        /**< Stop playing text, playComplete() will be notified, does not work if paused */
 
-    RC pause(  );
-    RC resume(  );
-  };
+    RC pause(  );                       /**< Pause the speech, if any */
+    RC resume(  );                      /**< Resume paused speech */
+
+		void setListener(in AnnounceListener listener);
+		void rmvListener(in AnnounceListener listener);    
+
+	};
 
 };
 
-- 
2.41.0.windows.1

